### DISCUSSION TITLE
Use correct Intel MKL config file for CMake

### DISCUSSION THREAD

**Discussion by caic99** [deepmodeling/abacus-develop#2299] [General]
### Describe Current Status and Possible Solution

Currently, we use a naive implementation to find MKL libraries in ABACUS. This might cause some problems on choosing the correct library when the environment is complicated, and might result in numerical errors. I strongly doubt this is the root cause of #1413 and #1121 . 
Since Intel OneAPI(2020+), the Intel MKL comes up with officially supported CMake config files. We should use it whenever possible, and take our implementation as a fall back when using legacy MKL versions.

**Comment by dyzheng** [comment]
Has this Issue been solved? @caic99

**Comment by caic99** [comment]
> Has this Issue been solved? @caic99

No. 
@baixiaokuang Any updates?

**Comment by mohanchen** [comment]
Move to discussion

**Comment by caic99** [comment]
Fixed by #4613 and following PRs.
